src/cargo/ops/cargo_compile.rs

```diff
@@ -834,7 +810,28 @@ fn list_rule_targets<'a>(
             }
         }
     }
-    Ok(result)
+    Ok(proposals)
+}
+
+fn list_all_targets<'a>(
+    packages: &[&'a Package],
+    is_expected_kind: fn(&Target) -> bool,
+    mode: CompileMode,
+) -> Vec<Proposal<'a>> {
+    let mut proposals = Vec::new();
+    for pkg in packages {
+        for target in pkg.targets() {
+            if is_expected_kind(target) {
+                proposals.push(Proposal {
+                    pkg,
+                    target,
+                    requires_features: false,
+                    mode,
+                });
+            }
+        }
+    }
+    proposals
 }
 
 /// Find the targets for a specifically named target.
```

Comment:
Did you consider also using `list_all_targets` inside `find_named_targets`? Or would adding the `requires_features` as a parameter be too much?

Reply:
Good idea